### DISCUSSION TITLE
Create a Plugin which can create and use a new Django Model and make new APIs

### DISCUSSION THREAD

**Discussion by inf3cti0n95** [saleor/saleor#8552] [Q&A]
I want to build a plugin, not modifying the core code, add a functionality, few more details to the existing models like Product and Customer.

Anyone has experience how to do this?

**Comment by karolkielecki** [comment]
@inf3cti0n95 Hi! What functionality would you like to add? What exactly would you like to achieve?

**Comment by inf3cti0n95** [comment]
@karolkielecki 
I'm adding a Model called Artist and Artwork, which has a relation with
Different products. Without breaking the core code.

**Comment by inf3cti0n95** [comment]
@karolkielecki 

I'm trying to achieve this in a more elegant way. Can a plugin in saelor create their own model and API routes

https://stackoverflow.com/questions/67586964/how-can-i-create-a-saleor-plugin-that-is-a-django-app

**Comment by lumleon** [comment]
@karolkielecki Hi, I also have a concern about the model extension. If I want to add models of Driver,  Car, DeliveryCost for a new plugin to track delivery cost of an order, how can we create these new tables and expose the api for use?

Thanks.

**Comment by korycins** [comment]
Hey, the new models/django's apps can be created as described on the stackoverflow. Currently, there is no possibility to extend graphql API via plugins. You could consider the usage of public/private metadata to pass data to Saleor over our API. You could also consider building an app and connect it with Saleor  with graphql federation.
https://docs.saleor.io/docs/3.0/developer/extending/apollo-federation#federation-support-in-saleor
https://www.apollographql.com/docs/federation/
https://github.com/preply/graphene-federation
